### DISCUSSION TITLE
Q-Tree with different child-key for different levels

### DISCUSSION THREAD

**Discussion by jamesrothwellhughes** [quasarframework/quasar#15735] [General - Components / Directives / etc] [answered]
Hi, I am using Prisma to fetch my object model to populate a tree.  Unfortunately, each level in the hierarchy has a different children key e.g. 
{
  name: "blah",
  locations: [
    name: "bleh",
    places: [
      {
        name: "somewhere"
      }
    ]
  ]
}

I'd like the tree to show all levels, but the child-key changes per level.

Any suggestions on how I can do this are very welcome!  Can I put a q-tree inside a q-tree node for example?

Many thanks,
James.
  ]

**Comment by dongwa** [comment, marked as answer]
You can preprocess your data before putting it into a qtree.
A simple example:

```js
const data = [
  {
    name: 'blah',
    locations: [
      {
        name: 'bleh',
        places: [
          {
            name: 'somewhere',
          },
        ],
      },
    ],
  },
]; // your data

function transformData(tar) {
  for (let child of tar) {
    for (let key in child) {
      const item = child[key];
      if (Array.isArray(item)) {
        // The conditions for judgment should be based on the actual situation. This is just a simple example.
        child.children = item;
        transformData(child.children);
      }
    }
  }
  return tar;
}

const treeList = transformData(data);
console.log('treeList', treeList);
// bind treeList to qtree

```
